zng/resolver/context.go

```diff
@@ -182,7 +175,7 @@ func (c *Context) newRecordType(id int, aliases []Alias) (*zng.TypeRecord, error
 	var columns []zng.Column
 	for _, alias := range aliases {
 		innerID := alias.Id
-		typ, err := c.lookupType(int(innerID))
+		typ, err := c.lookupTypeWithLock(int(innerID))
```

Comment:
but it looks like we dont have the lock?

Reply:
(same in newSetType and newVectorType)

Reply:
that is called from the marshaler so no problem not holding the lock and this code is going away so I'm not worried about the confusing name right now.

Reply:
ok